grug/types/src/coins.rs

```diff
@@ -191,33 +191,33 @@ impl Coins {
     }
 
     /// Insert a new coin to the `Coins`.
-    pub fn insert(&mut self, coin: Coin) -> StdResult<()> {
+    pub fn insert(&mut self, coin: Coin) -> StdResult<&mut Self> {
```

Comment:
put `&mut Self` in the return so that i can chain multiple `.insert` methods. the syntax looks a little better this way.